cpp/tests/search/search_test.cpp

```diff
@@ -1618,6 +1619,35 @@ TEST_F(SearchTest, contains_nullable_column_false_string)
   ASSERT_EQ(result, expect);
 }
 
+TEST_F(SearchTest, contains_dictionary)
+{
+  cudf::test::strings_column_wrapper column({"00", "00", "17", "17", "23", "23", "29"});
+  auto dict_col = cudf::dictionary::encode(column);
+  EXPECT_TRUE(cudf::contains(dict_col->view(), string_scalar{"23"}));
+  EXPECT_FALSE(cudf::contains(dict_col->view(), string_scalar{"28"}));
+
+  cudf::test::strings_column_wrapper needles({"00", "17", "23", "27"});
+  auto dict_needles = cudf::dictionary::encode(needles);
+  fixed_width_column_wrapper<bool> expect{1, 1, 1, 1, 1, 1, 0};
```

Comment:
Wondering if the definition of `contains(haystack, needles)` is right?
```Returns a new column of type bool identifying for each element of haystack column, if that element is contained in needles column.```
Each element of `haystack` is checked if it is present in `needles`!

This also seems incoherent. 
`contains(haystack, scalar needle )` returns single boolean. (size of second argument)
`contains(haystack, column needles)` returns a column of size `haystack` (size of first argument)

Reply:
This in a doxygen comment? That does sound backward.

Sorry, I'm not seeing what the _incoherent_ comment is referencing. Where are you seeing this?